ft-sdk/src/auth/mod.rs

```diff
@@ -116,11 +116,17 @@ pub fn ud(
         Err(e) => return Err(e),
     };
 
+    let email = data
+        .verified_emails
+        .first()
+        .cloned()
+        .unwrap_or_else(|| data.emails.first().cloned().unwrap());
+
     Ok(Some(ft_sys::UserData {
         id,
         identity: data.identity.clone(),
```

Comment:
This line feels wrong. Should we not fetch `identity` column from DB to populate this? Also should we use email as fallback if it is empty?

Reply:
We have two identities now: 1. the top level db column `identity`, a unique human readable identifier for the user (username/email) 2. provider dependent `identity` for each provider, e.g. a user via github will have identity as their github username but via email-password auth will have identity as their email. The idea was to favor 2. from the start but while implementing we introduced a top level db column (I may be wrong here).
The correct thing that I think to do here is remove the top-level `identity` to avoid confusion.

using emails as fallback is fine. There's another determinant `verified_email: bool` which the caller can use to check if the email that they're getting is a verified or an unverified one

Reply:
The reason I added top level identified as I wanted to have a “main provider”, and the identifier of the main provider will be the top level identifier.

Reply:
Makes sense. I'll change this field to store value of the top level `identity` :+1:

Reply:
done in https://github.com/FifthTry/ft-sdk/pull/44/commits/7fead720285c1eabad3bb1771d5eaeb159f1be24